src/main/java/org/spongepowered/api/conversation/ResponseContext.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.conversation;
+
+import org.spongepowered.api.text.Text;
+
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * An interface to allow differentiation between context for commands and
+ * conversations. Contains the common methods.
+ */
+public interface ResponseContext {
+
+    /**
+     * Gets all values for the given argument. May return an empty list if no
+     * values are present.
+     *
+     * @param key The key to get values for
+     * @param <T> the type of value to get
+     * @return the collection of all values
+     */
+    <T> Collection<T> getAll(String key);
+
+    /**
+     * Gets all values for the given argument. May return an empty list if no
+     * values are present.
+     *
+     * @param key The key to get values for
+     * @param <T> the type of value to get
+     * @return the collection of all values
+     */
+    <T> Collection<T> getAll(Text key);
+
+    /**
+     * Gets the value for the given key if the key has only one value.
+     *
+     * @param key the key to get
+     * @param <T> the expected type of the argument
+     * @return the argument
+     */
+    <T> Optional<T> getOne(String key);
```

Comment:
So, I just needed this class to abstract away from `CommandContext` as it has methods I did not really want accessible as they did not make sense in relation to conversations. The implementation of this is here: https://github.com/SpongePowered/SpongeAPI/blob/bleeding/src/main/java/org/spongepowered/api/command/args/CommandContext.java#L94

I did this, so `GenericArguments` and `CommandElement` could be used so that whole system does not need to be rewritten and is already familiar to users. Doing this also allows me to add support for tab completion once I get around to that.